homeassistant/components/iron_os/entity.py

```diff
@@ -9,21 +9,23 @@
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
 from .const import MANUFACTURER, MODEL
-from .coordinator import IronOSLiveDataCoordinator
+from .coordinator import IronOSBaseCoordinator
 
 
-class IronOSBaseEntity(CoordinatorEntity[IronOSLiveDataCoordinator]):
+class IronOSBaseEntity(CoordinatorEntity[IronOSBaseCoordinator]):
     """Base IronOS entity."""
 
     _attr_has_entity_name = True
 
     def __init__(
         self,
-        coordinator: IronOSLiveDataCoordinator,
+        coordinator: IronOSBaseCoordinator,
         entity_description: EntityDescription,
     ) -> None:
         """Initialize the sensor."""
-        super().__init__(coordinator)
+        super().__init__(
+            coordinator, context=getattr(entity_description, "set_key", None)
```

Comment:
why the getattr?

Reply:
only the number (input) entities have a set key

Reply:
any changes required here?

Reply:
I'd rather make it a parameter for the superclass than use getattr to dynamically fetch it